Detectors/TPC/reconstruction/include/TPCReconstruction/TrackTPC.h

```diff
@@ -0,0 +1,108 @@
+/// \file TrackTPC.h
+/// \brief Definition of the TPC Track
+/// \author Thomas Klemenz, TU Muenchen, thomas.klemenz@tum.de
+
+
+#ifndef ALICEO2_TPC_TRACKTPC
+#define ALICEO2_TPC_TRACKTPC
+
+#include "DetectorsBase/Track.h"
+
+#include "TPCSimulation/Cluster.h"
+
+#include <TClonesArray.h>
+
+namespace o2 {
+namespace TPC {
+
+/// \class TrackTPC
+/// This is the definition of the TPC Track Object
+
+
+class TrackTPC final : public o2::Base::Track::TrackParBase {
+  public:
+
+    /// Default constructor
+    TrackTPC();
+
+    /// Constructor, initializing values for x, alpha and an array with Y, Z, sin(phi), tg(lambda) and q/pT
+    /// \param x X of track evaluation
+    /// \param alpha track frame angle
+    /// \param std::array par contains Y, Z, sin(phi), tg(lambda) and q/pT
+    TrackTPC(float x,float alpha, const std::array<float, o2::Base::Track::kNParams> &par);
+
+    /// Constructor, initializing values for x,y and z in an array, momenta px, py and pz in an array
+    /// \param std::array xyz contains x, y and z coordinates of the vertex of the origin
+    /// \param std::array pxpypz contains momenta in x, y and z direction
+    /// \param sign sign of the charge of the particle
+    /// \param sectorAlpha false: angle of pT direction, true: angle of the sector from X, Y coordinate for r>1; angle of pT direction for r==0
+    TrackTPC(const std::array<float,3> &xyz,const std::array<float,3> &pxpypz, int sign, bool sectorAlpha=true);
+
+    /// Destructor
+    ~TrackTPC() = default;
+
```

Comment:
@wiechula : you have a plain pointer mClusterArray, the default d-tor will delete the pointer but not the object it points to. 
I think the best would be to use a smart pointer, which will delete also the pointed class, e.g.
std::unique_ptr<TClonesArray> mClusterArray

This will also comply with coding conventions (no bare new/delete), since then the creation of the array will become (in the TrackTPC c-tor)
mClusterArray =  std::unique_ptr<TClonesArray>(new TClonesArray("o2::TPC::Cluster"))
and the getter should become
TClonesArray *GetClusterArray()  const { return mClusterArray.get(); }